### PR TITLE
docs: add 'Dynamic Schema Creation' page

### PR DESCRIPTION
**Pull request by hassankhan**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Description of Changes Made (if issue reference is not provided)**

Create a single page to collect all dynamic schema creation documentation.

**Note to reviewers:** I think we could probably add some examples of dynamic schema creation using the security context too; are there any other examples of interesting dynamic schema creation problems that are worth documenting?

### CODE REVIEW

**Review by rpaik — APPROVED** [review]

File: docs/content/Schema/Dynamic-Schema-Creation.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  https://cube.dev/docs/schema-execution-environment#cube-js-globals-cube-and-others
+
+```javascript
+// schema/DynamicSchema.js
+const fetch = require('node-fetch');
+
+asyncModule(async () => {
+  const dynamicCubes = await (
+    await fetch('http://your-api-endpoint/dynamicCubes')
+  ).json();
```
Comment (keydunov) [inline docs/content/Schema/Dynamic-Schema-Creation.md:43]: Can we show an example of how this JSON should look like? So users would know how to format `/dynamicCubes`API.

File: docs/content/Schema/Dynamic-Schema-Creation.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  const {
+    authInfo: { deploymentId },
+  } = COMPILE_CONTEXT;
+
+  const dynamicCubes = await (
+    await fetch(`http://your-api-endpoint/dynamicCubes`)
+  ).json();
+
+  const allowedDimensions = await (
+    await fetch(`http://your-api-endpoint/dynamicDimensions/${deploymentId}`)
```
Comment (keydunov) [inline docs/content/Schema/Dynamic-Schema-Creation.md:102]: Should we use `tenantId` in the example here as well, same as before?

File: docs/content/Schema/Dynamic-Schema-Creation.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+---
+title: Dynamic Schema Creation
+permalink: /schema/dynamic-schema-creation
+category: Data Schema
+menuOrder: 2
+---
+
+Cube.js allows schemas to be created on-the-fly using a special
+[`asyncModule()`][ref-async-module] function only available in the [schema
```
Comment (igorlukanin) [inline docs/content/Schema/Dynamic-Schema-Creation.md:9]: It's really a very small detail but I'd like to ask whether we should add parentheses in the end of function names. I understand why we can do that but the drawback is that one might think that they are argument-less functions which is rarely true. WDYT, @hassankhan? (Anyway, regardless of the side you take :), it's worth documenting how we spell function names.)

Comment (hassankhan) [reply docs/content/Schema/Dynamic-Schema-Creation.md:9]: @igorlukanin Great point, I've add parentheses mainly to distinguish functions from variables/properties. I usually try to link out to the function's documentation too where possible. Happy to go either way though 😀

Comment (keydunov) [reply docs/content/Schema/Dynamic-Schema-Creation.md:9]: I believe it is a quite common notation for functions to display name with parentheses.

File: docs/content/Schema/Dynamic-Schema-Creation.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+// schema/DynamicSchema.js
+const fetch = require('node-fetch');
+import {
+  convertStringPropToFunction,
+  transformDimensions,
+  transformMeasures,
+} from './utils';
+
+asyncModule(async () => {
+  const dynamicCubes = await (
```
Comment (ovr) [inline docs/content/Schema/Dynamic-Schema-Creation.md:97]: @hassankhan Should we write somewhere pretty simple notice that JSON doesn't support serialization for functions?

Comment (hassankhan) [reply docs/content/Schema/Dynamic-Schema-Creation.md:97]: @ovr I tried to explain that on lines 23-27, perhaps I could be clearer?

**Comment by keydunov** [comment]
:shipit:

**Pull request merged**